enumerable.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(i, self[i])
+      i += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+    my_each do |x|
+      new_array << x if yield(x)
+    end
+    new_array
+  end
```

Comment:
* [ ] If no block is given (i.e check with block_given? at the beginning of the method) the method should return an Enumerator.